File: 53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        int maxSum = nums[0];
        int currentSum = 0;

        for (int num : nums) {
            if (currentSum < 0) {
                currentSum = 0;
            }
            currentSum += num;
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }
}

File: 53. Maximum Subarray.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        
        
        var currentSum = 0
        var result: Int = nums[0]
        
        for num in nums {
            if currentSum < 0 {
                currentSum = 0
            }
            currentSum += num
            result = max(result,currentSum)
        }
        
 
        return result
    }
}